### PR TITLE
fix: don't throttle duration change updates

### PR DESCRIPTION
**Pull request by BrandonOCasey**
## Description
Right now the `durationchange` event is throttled with the other two events, `timeupdate` and `loadedmetadata`. This means that only one of those events will trigger an update if they all occur within 25ms of each other. 

This functionality makes sense for `timeupdate` or `loadedmetadata` as those should not indicate a time update (even though they often do). 

For `durationchange` however it will always indicate a change in the duration, and we want to always update the display when it happens. Here is a scenario of how we could show duration incorrectly right now:

User is playing a video that has a postroll ad at the end. After the postroll ad their will be a `timeupdate`, and then a `durationchange` to signify that we are back in content. Then `ended` will fire, and no more events will happen.

The player will still show the duration of the ad, as the `durationchange` was eaten by the `timeupdate` that happened at nearly the same time.

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [x] If necessary, more likely in a feature request than a bug fix
  - [x] Change has been verified in an actual browser (Chome, Firefox, IE)
  - [ ] Example created ([starter template on JSBin](http://jsbin.com/axedog/edit?html,output))

### CODE REVIEW

File: src/js/control-bar/time-controls/time-display.js [hunk 1]
```diff
@@ -48,7 +48,7 @@ class TimeDisplay extends Component {
       'aria-live': 'off'
     }, Dom.createEl('span', {
       className: 'vjs-control-text',
-      textContent: this.localize(this.contentText_)
+      textContent: this.localize(this.controlText_)
```
Comment (BrandonOCasey) [inline src/js/control-bar/time-controls/time-display.js]: this was a typo, I could arguable pull this into its own pr if we want, but this is one of the causes of the broken unit tests.

File: src/js/control-bar/time-controls/time-display.js [hunk 2]
```diff
@@ -63,7 +63,7 @@ class TimeDisplay extends Component {
    * @private
    */
   updateTextNode_() {
-    if (this.textNode_) {
+    if (this.textNode_ && this.contentEl_ && this.contentEl_.contains(this.textNode_)) {
       this.contentEl_.removeChild(this.textNode_);
```
Comment (BrandonOCasey) [inline src/js/control-bar/time-controls/time-display.js]: IE11 throws in request animation frames when trying to remove a DOM node from a container when the container does not have the node. Might only be an issue for our unit tests, but I fixed it here.

File: src/js/control-bar/time-controls/time-display.js (hunk 1 above)
Comment (gkatsev) [reply src/js/control-bar/time-controls/time-display.js]: This is what's causing the issue on ie8? https://travis-ci.org/videojs/video.js/jobs/290639178#L1937-L1941

Comment (BrandonOCasey) [reply src/js/control-bar/time-controls/time-display.js]: im actually not sure what is causing it

File: src/js/component.js [hunk 3]
```diff
@@ -1239,9 +1239,7 @@ class Component {
     fn = Fn.bind(this, fn);
 
     const timeoutId = window.setTimeout(fn, timeout);
-    const disposeFn = function() {
-      this.clearTimeout(timeoutId);
-    };
+    const disposeFn = () => this.clearTimeout(timeoutId);
```
Comment (BrandonOCasey) [inline src/js/component.js]: This seemed a bit weird to me in a regular function we wouldn't have access to `this.clearTimeout`

File: src/js/component.js [hunk 4]
```diff
@@ -1302,9 +1300,7 @@ class Component {
 
     const intervalId = window.setInterval(fn, interval);
 
-    const disposeFn = function() {
-      this.clearInterval(intervalId);
-    };
+    const disposeFn = () => this.clearInterval(intervalId);
```
Comment (BrandonOCasey) [inline src/js/component.js]: fixed here as well

File: src/js/component.js (hunk 3 above)
Comment (gkatsev) [reply src/js/component.js]: we call the `dispose` handler with context as the component instance, so, it should have it.

Comment (gkatsev) [reply src/js/component.js]: at least, it should be. this is ok too.

File: src/js/control-bar/time-controls/time-display.js [hunk 5]
```diff
@@ -63,7 +63,12 @@ class TimeDisplay extends Component {
    * @private
    */
   updateTextNode_() {
-    if (this.textNode_) {
+    if (!this.contentEl_) {
+      return;
+    }
+
+    // ie8/9 don't support `el.contains` inside of setTimeout call.
```
Comment (gkatsev) [inline src/js/control-bar/time-controls/time-display.js]: weird

File: src/js/control-bar/time-controls/time-display.js [hunk 6]
```diff
@@ -63,7 +63,12 @@ class TimeDisplay extends Component {
...
+    if (this.textNode_ && this.textNode_.parentNode === this.contentEl_) {
```
Comment (gkatsev) [inline src/js/control-bar/time-controls/time-display.js]: what about just removing everything inside of `contentEl_`? It'll also fix an issue where if the page is translated in chrome using google translate, the text node will be wrapped in another element and this method will fail.
This could also just happen in a separate PR.

File: src/js/component.js [hunk 7]
```diff
@@ -1239,9 +1239,7 @@ class Component {
     fn = Fn.bind(this, fn);
 
     const timeoutId = window.setTimeout(fn, timeout);
```
Comment (gkatsev) [inline src/js/component.js]: I almost commented saying that this should be changed to `this.setTimeout` before realizing what this line is for 😆
